tqecjs/src/workspace/components/button.js

```diff
@@ -0,0 +1,50 @@
+import { Text, Container, Graphics } from 'pixi.js';
+import { DropShadowFilter } from '@pixi/filter-drop-shadow';
+
+export const button = (text, x, y, color = 0x000000) => {
+	// Create the button container
+	const button = new Container();
+	// Create the button text
+	const buttonText = new Text(text, {
+		fontFamily: 'Arial',
+		fontSize: 15,
+		// Adjust the font size based on the width and height of the button
+		fill: 0xffffff,
+		align: 'center',
+	});
+	buttonText.anchor.set(0.5);
+	buttonText.x = x;
+	buttonText.y = y;
+	// Create the button background
+	const buttonBackground = new Graphics();
+	buttonBackground.beginFill(color);
+	buttonBackground.drawRoundedRect(
+		buttonText.x - buttonText.getBounds().width / 2 - 15,
+		buttonText.y - buttonText.getBounds().height / 2 - 15,
+		buttonText.getBounds().width + 30,
+		buttonText.getBounds().height + 30
+	);
+
+	buttonBackground.endFill();
+
+	// Add effects
+	button.interactive = true;
+	button.eventMode = 'static';
+	// Add hover event
+	button.on('pointerover', () => {
+		buttonBackground.alpha = 0.5;
+	});
+	button.on('pointerout', () => {
+		buttonBackground.alpha = 1;
+	});
+
+	// Add shadow
+	buttonBackground.filters = [new DropShadowFilter()];
```

Comment:
This also creates a warning, but I'm unsure of an alternative
```
deprecation.ts:36 PixiJS Deprecation Warning: settings.FILTER_RESOLUTION is deprecated, use Filter.defaultResolutionDeprecated since v7.1.0
deprecation.ts:44     at ./node_modules/@pixi/filter-drop-shadow/dist/filter-drop-shadow.mjs (http://localhost:3000/static/js/bundle.js:87183:64)
```

Reply:
let's just keep this for now